File: 300-longest-increasing-subsequence/300-longest-increasing-subsequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const lengthOfLIS = (nums) => {
	let dp = new Array(nums.length).fill(1);
	let longest = 1;
	for (let i = 1; i < nums.length; i++) {
		for (let j = 0; j < i; j++) {
			if (nums[j] < nums[i]) {
				dp[i] = Math.max(dp[i], dp[j]+1);
				longest = Math.max(longest, dp[i]);
			}
		}
	}
	return longest;
}

File: 300-longest-increasing-subsequence/300-longest-increasing-subsequence.cs
public class Solution {
    public int LengthOfLIS(int[] nums) {
        var dp = new int[nums.Length];
        Array.Fill(dp, 1);
        for(var i = 0; i < nums.Length; i++)
        {
            for(var j = 0; j < i; j++)
            {
                if(nums[i] > nums[j])
                {
                    dp[i] = Math.Max(dp[i], 1 + dp[j]);
                }
            }
        }

        return dp.Max();
    }
}